bot/src/plugins/github_issues.cr

```diff
@@ -14,6 +27,16 @@ class GithubIssues
     return unless msg.channel?
     return unless config.repositories.has_key? msg.channel.name
     issues = msg.message.scan(PATTERN)
-    msg.reply issues.map {|issue| "https://github.com/#{config.repositories[msg.channel.name]}/issues/#{issue[1]}" }.join(" | ")
+    issues_data = issues.map {|issue| get_issue config.repositories[msg.channel.name], issue[1]}
+    msg.reply issues_data.compact.join(" | ")
+  end
+
+  private def get_issue repo, issue_id
+    api_response = HTTP::Client.get("https://api.github.com/repos/#{repo}/issues/#{issue_id}",
+      headers: HTTP::Headers{"Accept" => ["application/vnd.github.v3+json"], "User-Agent" => ["cbot_issue_fetch"]})
+    if api_response.status_code == 200
+      issue = Issue.from_json api_response.body
+      issue.to_s
```

Comment:
Actually just return the `Issue` object here, `.join` will use a single `IO` and call `to_s(io)`, avoiding a couple of intermediate strings.